### PR TITLE
Template collections fix

### PR DESCRIPTION
**Pull request by Ihromant**
Fixed multiple elements set and map for values with negative hashcode. Simplified code

### CODE REVIEW

File: classlib/src/main/java/org/teavm/classlib/java/util/TTemplateCollections.java [hunk 1]
```diff
@@ -201,32 +201,19 @@ public int size() {
             for (T element : data) {
                 Objects.requireNonNull(element);
 
-                int suggestedIndex = element.hashCode() % data.length;
+                int suggestedIndex = Math.abs(element.hashCode()) % data.length;
                 int index = suggestedIndex;
-                boolean found = false;
-                while (index < data.length) {
-                    T existingElement = table[index];
+                while (index < suggestedIndex + data.length) {
```
Comment (konsoletyper) [inline classlib/src/main/java/org/teavm/classlib/java/util/TTemplateCollections.java:206]: What's the purpose of this change?

Comment (Ihromant) [reply classlib/src/main/java/org/teavm/classlib/java/util/TTemplateCollections.java:206]: Code simplification and unification. Main issue was fixed by Math.abs(hashCode), but considered that it would be better to have fewer and clearer lines of code.

Comment (konsoletyper) [reply classlib/src/main/java/org/teavm/classlib/java/util/TTemplateCollections.java:206]: Please, revert this. I intentionally unswitched this loop manually to make it faster.

Comment (Ihromant) [reply classlib/src/main/java/org/teavm/classlib/java/util/TTemplateCollections.java:206]: Ok, will do it.
